erizo/src/test/NiceConnectionTest.cpp

```diff
@@ -0,0 +1,426 @@
+    EXPECT_CALL(*libnice, NiceAgentSetRelayInfo(_, _, _, _, _, _, _)).Times(0);
+
+    nice_connection = new erizo::NiceConnection(libnice_pointer,
+        erizo::VIDEO_TYPE,
+        "video",
+        "test_connection",
+        nice_listener,
+        2,
+        *ice_config);
+    nice_connection->start();
+  }
+
+  virtual void TearDown() {
+    delete nice_connection;
+    delete nice_listener;
+    delete ice_config;
+  }
+
+  boost::shared_ptr<erizo::LibNiceInterface> libnice_pointer;
+  MockLibNice* libnice;
+  MockNiceConnectionListener* nice_listener;
+  erizo::IceConfig* ice_config;
+  erizo::NiceConnection* nice_connection;
+  char* ufrag, * pass;
+};
+
+TEST_F(NiceConnectionStartTest, start_Configures_Libnice_With_Default_Config) {
+  char *ufrag = strdup("ufrag");
+  char *pass = strdup("pass");
+
+  EXPECT_CALL(*libnice, NiceAgentNew(_)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentAddStream(_, _)).Times(1).WillOnce(Return(1));
+  EXPECT_CALL(*libnice, NiceAgentGetLocalCredentials(_, _, _, _)).Times(1).
+    WillOnce(DoAll(SetArgPointee<2>(ufrag), SetArgPointee<3>(pass), Return(true)));
+  EXPECT_CALL(*libnice, NiceAgentAttachRecv(_, _, _, _, _, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentGatherCandidates(_, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentSetRemoteCredentials(_, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetPortRange(_, _, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetRelayInfo(_, _, _, _, _, _, _)).Times(0);
+
+  erizo::NiceConnection nice(libnice_pointer,
+      erizo::VIDEO_TYPE,
+      "video",
+      "test_connection",
+      nice_listener,
+      1,
+      *ice_config);
+  nice.start();
+}
+
+TEST_F(NiceConnectionStartTest, start_Configures_Libnice_With_Remote_Credentials) {
+  char *ufrag = strdup("ufrag");
+  char *pass = strdup("pass");
+
+  EXPECT_CALL(*libnice, NiceAgentNew(_)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentAddStream(_, _)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentGetLocalCredentials(_, _, _, _)).Times(1).
+    WillOnce(DoAll(SetArgPointee<2>(ufrag), SetArgPointee<3>(pass), Return(true)));
+  EXPECT_CALL(*libnice, NiceAgentAttachRecv(_, _, _, _, _, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentGatherCandidates(_, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentSetRemoteCredentials(_, _, _, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentSetPortRange(_, _, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetRelayInfo(_, _, _, _, _, _, _)).Times(0);
+
+  erizo::NiceConnection nice(libnice_pointer,
+      erizo::VIDEO_TYPE,
+      "video",
+      "test_connection",
+      nice_listener,
+      1,
+      *ice_config,
+      "ufrag",
+      "pass");
+  nice.start();
+}
+
+TEST_F(NiceConnectionStartTest, start_Configures_Libnice_With_Port_Range) {
+  char *ufrag = strdup("ufrag");
+  char *pass = strdup("pass");
+
+  ice_config->minPort = 10;
```

Comment:
consider using constants for this arbitrary numbers, it improves readability. For instance: kArbitraryMinPort and kArbitraryMaxPort. Or use ice_config->minPort below as you do in other tests.